pkg/pillar/agentlog/agentlog.go

```diff
@@ -396,3 +396,24 @@ func CheckMaxTime(agentName string, start time.Time) {
 			fn, line, agentName, elapsed/time.Second)
 	}
 }
+
+// CheckMaxTimeTopic verifies if the time for a call has exeeded a reasonable
+// number.
+func CheckMaxTimeTopic(agentName string, topic string, start time.Time) {
+	errTime := errorTime
+	warnTime := warningTime
+	if agentName == "nim" {
+		errTime = errorTimeNim
+		warnTime = warningTimeNim
+	}
+	elapsed := time.Since(start)
+	if elapsed > errTime {
+		_, fn, line, _ := runtime.Caller(1)
```

Comment:
Is the purpose of `CheckMaxTimeTopic` not to avoid the whole `runtime.Caller(1)` magic, and instead by purely functional, i.e. "I am a function that will check the time you passed, and if it is a problem, will log a message with the message parameters you gave me"?

Reply:
Issue is that we have several call sites from different select loops and we might want to know which one caused the issue. Alternatively we handle that by making sure we have log.Info messages each time we move from one select loop.

Reply:
Understood. It does make it non-functional, which can trip people up (as it did me). More relevantly, though, is that line numbers are useful for crashes, that are resolved by developers; not so much for performance issues, where you care what caused X to spend so much time, which is more operational.

Reply:
FWIW I removed this use of Caller by making pubsub and other callers specify the topic aka collection which had the time issue.